wls-wahlvorstand-service/src/main/java/de/muenchen/oss/wahllokalsystem/wahlvorstandservice/service/wahlvorstand/WahlvorstandService.java

```diff
@@ -0,0 +1,196 @@
+        });
+
+        KonfigurierterWahltagModel konfigurierterWahltagModel = konfigurierterWahltagClient.getKonfigurierterWahltag();
+        return Optional.of(persistWahlvorstand(fallbackWahlvorstand, konfigurierterWahltagModel));
+    }
+
+    private WahlvorstandModel persistWahlvorstand(WahlvorstandModel wahlvorstand, KonfigurierterWahltagModel konfigurierterWahltagModel) {
+        if (wahlvorstand == null || wahlvorstand.wahlvorstandsmitglieder().isEmpty()) {
+            return null;
+        }
+
+        val wahlvorstandDB = wahlvorstandRepository.findById(wahlvorstand.wahlbezirkID());
+        if (wahlvorstandDB.isPresent()) {
+            try {
+                wahlvorstandRepository.save(populateFunktionsnameOffline(wahlvorstandModelMapper.toEntity(wahlvorstand), wahlvorstandDB.get()));
+            } catch (IllegalStateException ex) {
+                wahlvorstandRepository.save(populateFunktionsnameOnline(wahlvorstandModelMapper.toEntity(wahlvorstand), konfigurierterWahltagModel));
+            }
+        } else {
+            wahlvorstandRepository.save(populateFunktionsnameOnline(wahlvorstandModelMapper.toEntity(wahlvorstand), konfigurierterWahltagModel));
+        }
+        return wahlvorstand;
+    }
+
+    private Wahlvorstand populateFunktionsnameOffline(Wahlvorstand wahlvorstand, Wahlvorstand wahlvorstandDB) throws IllegalStateException {
+        List<Wahlvorstandsmitglied> collect = wahlvorstand.getWahlvorstandsmitglieder().stream()
+                .map(mitglied -> populateWahlvorstandsmitgliedFunktionsnameOffline(mitglied, wahlvorstandDB))
+                .toList();
+        wahlvorstand.setWahlvorstandsmitglieder(collect);
+        return wahlvorstand;
+    }
+
+    private Wahlvorstandsmitglied populateWahlvorstandsmitgliedFunktionsnameOffline(Wahlvorstandsmitglied mitglied, Wahlvorstand wahlvorstandDB)
+            throws IllegalStateException {
+        Wahlvorstandsmitglied mitgliedDB = wahlvorstandDB.getWahlvorstandsmitglieder().stream()
+                .filter(wahlvorstandsmitglied -> wahlvorstandsmitglied.getFunktion().equals(mitglied.getFunktion()))
+                .findFirst()
+                .orElse(null);
+        if (mitgliedDB != null) {
+            mitglied.setFunktionsname(mitgliedDB.getFunktionsname());
+        } else {
+            throw new IllegalStateException("Bisher unbekannte Funktion in Wahlvorstand gefunden. Muss neu gemapt werden.");
+        }
+        return mitglied;
+    }
+
+    private Wahlvorstand populateFunktionsnameOnline(Wahlvorstand wahlvorstand, KonfigurierterWahltagModel wahltagModel) {
+        WahlbezirkArt wahlbezirkArt = getWahlbezirkArt();
+
+        List<WahlModel> wahlen = wahlenClient.getWahlen(wahltagModel);
+        if (wahlen == null) throw exceptionFactory.createFachlicheWlsException(ExceptionConstants.BASISDATEN_ANTWORT_NULL);
+        Wahlart zuerstAuszuzaehlendeWahl = wahlen.get(0).wahlart();
```

Comment:
die funktionalität könnte man auch and den Client auslagern. Der Wahlvorstandsservice sollte keine Kenntnisse benötigen wie man die zuerst auszuzählende Wahl bestimmt.

Reply:
Neues Issue: #560